### PR TITLE
session tinkering, mostly logging

### PR DESCRIPTION
**Pull request by fredrikt**

### CODE REVIEW

File: src/eduid_common/session/redis_session.py [hunk 1]
```diff
@@ -229,24 +230,27 @@ def __init__(
...
         self.raise_on_unknown = raise_on_unknown
-        self.app_secret = app_secret
         # encrypted data loaded from redis, used to avoid clobbering concurrent updates to the session
         self._raw_data: Optional[str] = None
 
-        if cookie_val:
-            self.token = SessionCookie.from_cookie(cookie_val, app_secret=self.app_secret)
-        else:
-            self.token = SessionCookie.new(app_secret=self.app_secret)
+        #        if cookie_val:
```
Comment (johanlundberg) [inline src/eduid_common/session/redis_session.py:241]: This should go, right?

File: src/eduid_common/session/session_cookie.py [hunk 2]
```diff
@@ -112,6 +112,11 @@ def _verify_session_id(cls, session_id: bytes, signing_key: bytes, signature: by
 
         return bool(not invalid_bits)
 
+    def derive_key(self, app_secret: str, usage: str, size: int) -> bytes:
```
Comment (johanlundberg) [inline src/eduid_common/session/session_cookie.py:115]: isn't this confusing with the derive_key below in the outer scope?

File: src/eduid_common/session/redis_session.py (hunk 1 above)
Comment (fredrikt) [reply src/eduid_common/session/redis_session.py]: right

File: src/eduid_common/session/session_cookie.py (hunk 2 above)
Comment (fredrikt) [reply src/eduid_common/session/session_cookie.py:115]: At this time, I didn't want to declare derive_key to be _only_ related to SessionCookie, but I wanted to encapsulate the complexity of juggling the types of session_id into SessionCookie. Good or bad I do not know.

**Pull request merged**